diff --git a/db/store.go b/db/store.go
index bf97033..5039567 100644
--- a/db/store.go
+++ b/db/store.go
@@ -282,17 +282,17 @@
 
 		for res.NextResultSet(ctx) {
 			for res.NextRow() {
-				var path string
+				var path *string
 				var accessType *string
-				err = res.Scan(&path, &accessType)
+				err = res.ScanNamed(
+					named.Optional("path", &path),
+					named.Optional("access_type", &accessType),
+				)
 				if err != nil {
 					return err
 				}
-				if accessType != nil {
-					permissions[path] = *accessType
+				if path != nil && accessType != nil {
+					permissions[*path] = *accessType
 				}
 			}
 		}